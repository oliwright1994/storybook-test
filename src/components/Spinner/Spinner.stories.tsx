import createStory from '@bright/storybook/createStory'
import { select } from '@storybook/addon-knobs'
import React from 'react'

import Spinner from '.'

createStory('COMPONENTS | Spinner', [
  {
    label: 'standard',
    jsx: () => {
      const size = select('Size', ['xl', 'lg', 'md', 'sm', 'xs'], 'lg')

      return <Spinner size={size} />
    },
  },
])
