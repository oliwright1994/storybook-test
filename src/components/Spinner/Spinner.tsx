import { Spinner as ChakraSpinner } from '@chakra-ui/core'
import React, { FC } from 'react'

interface ISpinner {
  size?: 'xl' | 'lg' | 'md' | 'sm' | 'xs'
}

const Spinner: FC<ISpinner> = ({ size = 'md' }) => (
  <ChakraSpinner thickness="2px" speed="0.8s" emptyColor="transparent" color="blackAlpha.500" size={size} />
)

export default Spinner
