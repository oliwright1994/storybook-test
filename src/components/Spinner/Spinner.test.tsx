import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import Spinner from '.'

describe('<Spinner />', () => {
  it('should render as expected', () => {
    const { asFragment } = renderWithBrightTheme(<Spinner size="md" />)

    expect(asFragment()).toMatchSnapshot()
  })
})
