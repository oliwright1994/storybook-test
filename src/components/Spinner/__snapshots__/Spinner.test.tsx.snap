// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render as expected 1`] = `
<DocumentFragment>
  <div
    class="css-lhw0gv"
  >
    <div
      class="css-1gy22ay"
    >
      Loading...
    </div>
  </div>
</DocumentFragment>
`;
